File: src/types/table-editor.ts
export interface ColumnDefinition {
  name: string
  dataType: string
  length: string | null
  nullable: boolean
  isPrimaryKey: boolean
  autoIncrement: boolean
  defaultValue: string | null
  onUpdate: string | null
  comment: string | null
}

export interface IndexDefinition {
  name: string
  columns: string[]
  indexType: string // "PRIMARY", "UNIQUE", "INDEX", "FULLTEXT"
}

export interface ForeignKeyDefinition {
  name: string
  columns: string[]
  refTable: string
  refColumns: string[]
  onDelete: string | null
  onUpdate: string | null
}

export interface TableDefinition {
  name: string
  database: string
  columns: ColumnDefinition[]
  indexes: IndexDefinition[]
  foreignKeys: ForeignKeyDefinition[]
  engine: string | null
  charset: string | null
  collation: string | null
  comment: string | null
}

export interface ColumnChange {
  changeType: string
  column: ColumnDefinition
  oldName: string | null
  afterColumn: string | null
}

export interface IndexChange {
  changeType: string
  index: IndexDefinition
}

export interface TableAlteration {
  database: string
  table: string
  columnChanges: ColumnChange[]
  indexChanges: IndexChange[]
  newName: string | null
  newComment: string | null
  newEngine: string | null
  newCharset: string | null
}

export interface DdlResult {
  sql: string
  statements: string[]
}

export interface TableDetail {
  name: string
  columns: ColumnDefinition[]
  indexes: IndexDefinition[]
  foreignKeys: ForeignKeyDefinition[]
  engine: string | null
  charset: string | null
  collation: string | null
  comment: string | null
}

File: src-tauri/src/models/table_editor.rs
use serde::{Deserialize, Serialize};

/// Column definition for creating/altering tables
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ColumnDefinition {
    pub name: String,
    pub data_type: String,
    pub length: Option<String>,
    pub nullable: bool,
    pub is_primary_key: bool,
    pub default_value: Option<String>,
    pub auto_increment: bool,
    pub on_update: Option<String>,
    pub comment: Option<String>,
}

/// Index definition
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct IndexDefinition {
    pub name: String,
    pub columns: Vec<String>,
    pub index_type: String, // "PRIMARY", "UNIQUE", "INDEX", "FULLTEXT"
}

/// Foreign key definition
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ForeignKeyDefinition {
    pub name: String,
    pub columns: Vec<String>,
    pub ref_table: String,
    pub ref_columns: Vec<String>,
    pub on_delete: Option<String>,
    pub on_update: Option<String>,
}

/// Full table definition for CREATE TABLE
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct TableDefinition {
    pub name: String,
    pub database: String,
    pub columns: Vec<ColumnDefinition>,
    pub indexes: Vec<IndexDefinition>,
    pub foreign_keys: Vec<ForeignKeyDefinition>,
    pub engine: Option<String>,
    pub charset: Option<String>,
    pub collation: Option<String>,
    pub comment: Option<String>,
}
/// Column change for ALTER TABLE
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ColumnChange {
    pub change_type: String, // "add", "modify", "drop", "rename"
    pub column: ColumnDefinition,
    pub old_name: Option<String>,
    pub after_column: Option<String>,
}

/// Index change for ALTER TABLE
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct IndexChange {
    pub change_type: String, // "add", "drop"
    pub index: IndexDefinition,
}

/// Table alteration definition
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct TableAlteration {
    pub database: String,
    pub table: String,
    pub column_changes: Vec<ColumnChange>,
    pub index_changes: Vec<IndexChange>,
    pub new_name: Option<String>,
    pub new_comment: Option<String>,
    pub new_engine: Option<String>,
    pub new_charset: Option<String>,
}

/// Result of DDL generation
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct DdlResult {
    pub sql: String,
    pub statements: Vec<String>,
}

/// Table detail info (for loading existing table into editor)
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct TableDetail {
    pub name: String,
    pub columns: Vec<ColumnDefinition>,
    pub indexes: Vec<IndexDefinition>,
    pub foreign_keys: Vec<ForeignKeyDefinition>,
    pub engine: Option<String>,
    pub charset: Option<String>,
    pub collation: Option<String>,
    pub comment: Option<String>,
}
